client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -119,12 +163,22 @@ public ReaderGroupConfig build() {
            checkArgument(startingStreamCuts != null && startingStreamCuts.size() > 0,
                    "Stream names that the reader group can read from cannot be empty");
 
-           startingStreamCuts.forEach((s, streamCut) -> {
+           //endStreamCuts is an optional configuration. Initialize if it is null.
+           if (endingStreamCuts == null) {
+               endingStreamCuts = Collections.emptyMap();
+           }
+
+           Consumer<Map<Stream, StreamCut>> validateStreamCuts = map -> map.forEach((s, streamCut) -> {
                if (!streamCut.equals(StreamCut.UNBOUNDED)) {
                    checkArgument(s.equals(streamCut.asImpl().getStream()));
                }
            });
-           return new ReaderGroupConfig(groupRefreshTimeMillis, automaticCheckpointIntervalMillis, startingStreamCuts);
+           //validate start and end StreamCuts
+           validateStreamCuts.accept(startingStreamCuts);
```

Comment:
It also should validate that start < end

Reply:
Improved the validation. (Validation without invoking controller has been added.)